packages/gatsby/cache-dir/navigation.js

```diff
@@ -97,10 +97,16 @@ const navigate = (to, options = {}) => {
   })
 }
 
-function shouldUpdateScroll(prevRouterProps, { location: { pathname } }) {
+function shouldUpdateScroll(prevRouterProps, routerProps) {
+  const {
+    location: { pathname, hash },
+  } = routerProps
   const results = apiRunner(`shouldUpdateScroll`, {
     prevRouterProps,
+    // `pathname` for backwards compatibility
     pathname,
+    routerProps,
```

Comment:
I'm scared about `history` being passed here, same as I'm not a fan of `location.replace`/`location.reload`. 

I wouldn't expose functionality that could be abused outside the normal loop of things and potentially causing unexpected results.